backend/src/macros.rs

```diff
@@ -0,0 +1,84 @@
+// NOTE: might be better to declare API as global constant, but the downside of this is
+// that we need to pass around this variable alongside with macro call
+// lazy_static! {
+//     pub static ref API: OpenApi = ApiDoc::openapi();
+// }
+
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum RouteError {
+    #[error("No route found in OpenAPI scheme")]
+    NoRoute,
+    #[error("No method found for specified route in OpenAPI scheme")]
+    NoMethod,
+    #[error("No `operation_id` found")]
+    NoOperation,
+    #[error("OpenAPI's `operation_id` doesn't match handler's name")]
+    NoMatch,
+}
+
+// TODO: add all axum's routing options (wildcards, any, on, service routing) support
+/// Check if the following route corresponds with `OpenAPI` declaration
+/// Relies on `operation_id` field, should NOT be changed on new route declaration
+#[macro_export]
+macro_rules! new_api_route {
+    ($router:expr, $route:literal, get($func:ident)) => {{
+        new_api_route!($route, $func, Get).map(|_| $router.route($route, get($func)))
+    }};
+    ($router:expr, $route:literal, post($func:ident)) => {{
+        new_api_route!($route, $func, Post).map(|_| $router.route($route, post($func)))
+    }};
+    ($router:expr, $route:literal, delete($func:ident)) => {{
+        new_api_route!($route, $func, Delete).map(|_| $router.route($route, delete($func)))
+    }};
+    ($router:expr, $route:literal, put($func:ident)) => {{
+        new_api_route!($route, $func, Put).map(|_| $router.route($route, put($func)))
+    }};
+    ($router:expr, $route:literal, head($func:ident)) => {{
+        new_api_route!($route, $func, Head).map(|_| $router.route($route, head($func)))
+    }};
+    ($router:expr, $route:literal, options($func:ident)) => {{
+        new_api_route!($route, $func, Options).map(|_| $router.route($route, options($func)))
+    }};
+    ($router:expr, $route:literal, patch($func:ident)) => {{
+        new_api_route!($route, $func, Patch).map(|_| $router.route($route, patch($func)))
+    }};
+    ($router:expr, $route:literal, trace($func:ident)) => {{
+        new_api_route!($route, $func, Trace).map(|_| $router.route($route, trace($func)))
+    }};
+    ($router:expr, $route:literal, connect($func:ident)) => {{
+        new_api_route!($route, $func, Connect).map(|_| $router.route($route, connect($func)))
+    }};
+    ($route:literal, $func:ident, $method:ident) => {{
+        || -> Result<(), RouteError> {
+            #[cfg(not(feature = "swagger"))]
+            return Ok(());
+            #[cfg(feature = "swagger")]
```

Comment:
I think, this check is not needed in release builds. Please, add additional condition to `cfg` to eliminate checks in release (https://stackoverflow.com/questions/39204908/how-to-check-release-debug-builds-using-cfg-in-rust)

Reply:
I'm a little confused here.
1. Should every `cfg` check be modified or this particular one? 
2. Should it be like this:

```rust
    #[cfg(any(not(debug_assertions), not(feature = "swagger")))] // Release build OR swagger feature is disabled
    {... release behaviour... }
    ...
```
Or like this:
```rust
    #[cfg(not(debug_assertions))] // If this release build
    {... release behaviour... }
    ...
```
?

Reply:
`#[cfg(all(feature = "swagger", debug_assertions))]`. In this way the check will be run only in debug build with `swagger` enabled as a feature